packages/namers/default/src/DefaultNamer.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import {Namer} from '@parcel/plugin';
+import crypto from 'crypto';
+import path from 'path';
+
+const COMMON_NAMES = new Set(['index', 'src', 'lib']);
+const DEFAULT_DIST_DIR = 'dist';
+
+export default new Namer({
+  name(bundle) {
+    // If the bundle has an explicit file path given (e.g. by a target), use that.
+    if (bundle.filePath) {
+      // TODO: what about multiple assets in the same dep?
+      // e.g. input is a Vue file, output is JS + CSS
+      // which is defined as a target in package.json?
```

Comment:
Any ideas how we could tell which one is the main one? Should we take the specified filename and just replace the extension with the bundle type?

Reply:
Yeah, I think that works.